roles/lib_openshift/src/lib/base.py

```diff
@@ -22,6 +22,29 @@ def __init__(self,
         self.verbose = verbose
         self.kubeconfig = Utils.create_tmpfile_copy(kubeconfig)
         self.all_namespaces = all_namespaces
+        self.oc_binary = self._locate_oc_binary()
+
+    @staticmethod
+    def _locate_oc_binary():
+        ''' Find and return oc binary file '''
+        # https://github.com/openshift/openshift-ansible/issues/3410
+        # oc can be in /usr/local/bin in some cases, but that may not
+        # be in $PATH due to ansible/sudo
+        paths = os.environ.get("PATH", os.defpath).split(os.pathsep) + ['/usr/local/bin', os.path.expanduser('~/bin')]
+        paths = list(set(paths))
+
+        oc_binary = 'oc'
+        if six.PY2:
```

Comment:
This type of code raises a red flag on my brain. It is dangerous to carry different implementations for Python 3 and Python 2, specially without good automated tests around it.

Couldn't we resist the temptation to using `shutil.which` from Python 3 and have a single implementation that works for both Py2/3?